Helper/Data.php

```diff
@@ -175,13 +154,22 @@ private function updateSentStatus($quoteId)
     public function getGeneralConfig($code, $storeId = null)
     {
         $tab = 'general';
-        if (in_array($code, self::SUBSCRIBE_SETTINGS, true)) {
+        if (in_array(
```

Comment:
What if you do something like this to improve code readability:
```php
class Helper
{
    const TAB_SUBSCRIBE = 'subscribe';
    const TAB_SYNC = 'sync';
    const TAB_ABANDONED = 'abandoned';

    protected $_settingToTabMapping = [
        'enableNewsletterSubscriptions' => self::TAB_SUBSCRIBE,
        'enableCaptcha' => self::TAB_SUBSCRIBE,
        'captchaType' => self::TAB_SUBSCRIBE,
        'captchaApiKey' => self::TAB_SUBSCRIBE,
        'captchaApiSecret' => self::TAB_SUBSCRIBE,
        'fields' => self::TAB_SYNC,
        'frequency' => self::TAB_SYNC,
        'enableCronSync' => self::TAB_SYNC,
        'autoresponderId' => self::TAB_ABANDONED,
        'syncTime' => self::TAB_ABANDONED,
        'productfields' => self::TAB_ABANDONED,
        'enableAbandonedCart' => self::TAB_ABANDONED,
    ];

    // ...

    public function getGeneralConfig($code, $storeId = null) {
        $tab = in_array($code, $this->_settingToTabMapping, true) ? $this->_settingToTabMapping[$code] : 'general';
        return trim($this->getConfigValue(self::XML_PATH . $tab . '/' . $code, $storeId));
    }

    // ...
}
```

or

```php
class Helper
{
    protected $_subscribeSettings = [
        'enableNewsletterSubscriptions',
        'enableCaptcha',
        'captchaType',
        'captchaApiKey',
        'captchaApiSecret',
    ];

    protected $_syncSettings = ['fields', 'frequency', 'enableCronSync'];

    protected $_abandonedSettings = ['autoresponderId', 'syncTime', 'productfields', 'enableAbandonedCart'];

    // ...

    public function getGeneralConfig($code, $storeId = null) {
        $tab = 'general';
        if (in_array($code, $this->_subscribeSettings, true)) {
            $tab = 'subscribe';
        }
        elseif (in_array($code, $this->_syncSettings, true)) {
            $tab = 'sync';
        }
        elseif (in_array($code, $this->_abandonedSettings, true)) {
            $tab = 'abandoned';
        }

        return trim($this->getConfigValue(self::XML_PATH . $tab . '/' . $code, $storeId));
    }

    // ...
}
```

or

```php
class Helper
{
    // ...

    public function getGeneralConfig($code, $storeId = null) {
        $subscribeSettings = [
            'enableNewsletterSubscriptions',
            'enableCaptcha',
            'captchaType',
            'captchaApiKey',
            'captchaApiSecret',
        ];
        $syncSettings = ['fields', 'frequency', 'enableCronSync'];
        $abandonedSettings = ['autoresponderId', 'syncTime', 'productfields', 'enableAbandonedCart'];

        $tab = 'general';
        if (in_array($code, $subscribeSettings, true)) {
            $tab = 'subscribe';
        }
        elseif (in_array($code, $syncSettings, true)) {
            $tab = 'sync';
        }
        elseif (in_array($code, $abandonedSettings, true)) {
            $tab = 'abandoned';
        }

        return trim($this->getConfigValue(self::XML_PATH . $tab . '/' . $code, $storeId));
    }

    // ...
}
```

Would any of the examples conflict with Magento code anaylzer?

Reply:
I will try to go with the second option as it puts field values to the top of the file.
And according to PSR2, I should omit underscore at the beginning of variable names.

>Property names SHOULD NOT be prefixed with a single underscore to indicate protected or private visibility.

Reply:
Yeah, sure. Make every change necessary to make the code Magento compliant.